File: pass_Generator.js
// This program is contributed by Gaurav Mandal

function generatePassword(length = 12, useLowercase = true, useUppercase = true, useDigits = true, useSpecialChars = true) {
    let characters = '';

    if (useLowercase) {
        characters += 'abcdefghijklmnopqrstuvwxyz';
    }
    if (useUppercase) {
        characters += 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
    }
    if (useDigits) {
        characters += '0123456789';
    }
    if (useSpecialChars) {
        characters += '!@#$%^&*()_-+=<>?';
    }

    if (!(useLowercase || useUppercase || useDigits || useSpecialChars)) {
        throw new Error("At least one character set (lowercase, uppercase, digits, special chars) must be selected.");
    }

    let password = '';
    for (let i = 0; i < length; i++) {
        const randomIndex = Math.floor(Math.random() * characters.length);
        password += characters.charAt(randomIndex);
    }

    return password;
}

// Example usage:
const password = generatePassword(16, true, true, true, true);
console.log(password);

// This code is contributed by Gaurav Mandall (https://www.linkedin.com/in/gauravmandall/)

File: pass_Generator.rs
// This program is contributed by Gaurav Mandal

use rand::Rng;
use std::iter;

fn generate_password(length: usize, use_lowercase: bool, use_uppercase: bool, use_digits: bool, use_special_chars: bool) -> String {
    let mut characters = String::new();

    if use_lowercase {
        characters += "abcdefghijklmnopqrstuvwxyz";
    }
    if use_uppercase {
        characters += "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    }
    if use_digits {
        characters += "0123456789";
    }
    if use_special_chars {
        characters += "!@#$%^&*()_-+=<>?";
    }

    if !(use_lowercase || use_uppercase || use_digits || use_special_chars) {
        panic!("At least one character set (lowercase, uppercase, digits, special chars) must be selected.");
    }

    let password: String = iter::repeat_with(|| {
        let random_index = rand::thread_rng().gen_range(0..characters.len());
        characters.chars().nth(random_index).unwrap()
    })
    .take(length)
    .collect();

    password
}

fn main() {
    let password = generate_password(16, true, true, true, true);
    println!("{}", password);
}

// This code is contributed by Gaurav Mandall (https://www.linkedin.com/in/gauravmandall/)